service/controller/resource/cleanupendpointips/create.go

```diff
@@ -149,28 +154,58 @@ func removeFromEndpointAddressList(addresses []corev1.EndpointAddress, indexesTo
 	return newAddresses
 }
 
+func controlPlanePodForTCNode(node corev1.Node, pods []corev1.Pod) (corev1.Pod, error) {
+	for _, pod := range pods {
+		if pod.Name == node.Name {
+			return pod, nil
+		}
+	}
+	// Unless there is a race condition where the Pods are modified
+	// after being checked by podsEqualNodes(), this should never be reached
```

Comment:
I do not know if this is true.
```suggestion
	// after being checked by podsEqualNodes(), this should never be reached.
```

Reply:
When `controlPlanePodForTCNode` is called after `podsEqualNodes`, it is true because `podsEqualNodes` asserts that each Node has a same-named Pod and we don't call this function if that condition isn't met. If this were to be called in a different context, this would not necessarily be the case, but that's what the error is for.